docroot/profiles/custom/cgov_site/modules/custom/cgov_core/cgov_core.module

```diff
@@ -207,6 +207,14 @@ function cgov_core_form_alter(&$form, FormStateInterface $form_state, $form_id)
       unset($form['actions']['reset_alphabetical']);
     }
   }
+
+  if ($form_id === 'user_form') {
+    if (isset($form['account']['roles'])) {
+      $user = \Drupal::currentUser();
+      $form['account']['roles']['#access'] = $form['account']['roles']['#options'] &&
+        ($user->hasPermission('administer permissions') || $user->hasPermission('edit user roles'));
+    }
+  }
```

Comment:
This needs a comment to explain what it's doing and why.

Also, the hook's comment block is no longer correct.